Src/GoogleApis.Auth.DotNet4/OAuth2/GoogleCredential.cs

```diff
@@ -0,0 +1,299 @@
+        /// screen. Nothing additional is required.</description> 
+        /// </item> 
+        /// <item> 
+        /// <description>ServiceAccountCredentials does not have scopes built-in by default. Caller should 
+        /// invoke <see cref="Google.Apis.Auth.OAuth2.IScopableCredential.CreateScoped"/> to add scopes to the Credential.</description> 
+        /// </item> 
+        /// </list>
+        /// </summary>
+        public virtual bool IsCreateScopedRequired
+        {
+            get { return false; }
+        }
+
+        /// <summary>If the credential supports scopes, creates a copy with the specified scopes, otherwise return the same instance.</summary>
+        public virtual GoogleCredential CreateScoped(IEnumerable<string> scopes)
+        {
+            return this;
+        }
+
+        #region IConfigurableHttpClientInitializer
+
+        void IConfigurableHttpClientInitializer.Initialize(ConfigurableHttpClient httpClient)
+        {
+            Initialize(httpClient);
+        }
+
+        #endregion
+
+        #region ITokenAccess
+
+        TokenResponse ITokenAccess.Token
+        {
+            get { return Token; }
+        }
+
+        Task<bool> ITokenAccess.RequestAccessTokenAsync(CancellationToken taskCancellationToken)
+        {
+            return RequestAccessTokenAsync(taskCancellationToken);
+        }
+
+        #endregion
+
+        /// <summary>Provides access to the underlying credential object</summary>
+        internal abstract object UnderlyingCredential { get; }
+
+        // We're explicitly implementing all the interfaces to only expose the members user actually
+        // needs to see. Because you cannot make explicit interface implementors abstract, they are redirecting
+        // to the following protected abstract members.
+        protected abstract void Initialize(ConfigurableHttpClient httpClient);
+
+        protected abstract TokenResponse Token { get; }
+
+        protected abstract Task<bool> RequestAccessTokenAsync(CancellationToken taskCancellationToken);
+
+        #region Factory methods
+
+        internal static GoogleCredential FromCredential(ComputeCredential credential)
+        {
+            return new ComputeGoogleCredential(credential);
+        }
+
+        internal static GoogleCredential FromCredential(ServiceAccountCredential credential)
+        {
+            return new ServiceAccountGoogleCredential(credential);
+        }
+
+        internal static GoogleCredential FromCredential(UserCredential credential)
+        {
+            return new UserGoogleCredential(credential);
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Wraps <c>ComputeCredential</c> as <c>GoogleCredential</c>.
+        /// </summary>
+        internal class ComputeGoogleCredential : GoogleCredential
+        {
+            private readonly ComputeCredential credential;
```

Comment:
Cool, you already have it  ITokenAccess, so maybe we can change my classes as well.
TODO. Not urgent. But it will make your code easier, right?

Or that's what we are trying to avoid here? Not changing my classes at all?